File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows):
        ans = []
        for i in range(numRows):
            v = [0]*(i+1)
            for j in range(i+1):
                if j==0 or j==i:
                    v[j] = 1
                else:
                    v[j] = ans[i-1][j-1] + ans[i-1][j]
            ans.append(v)
        return ans

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>>ans;
        for(int i=0;i<n;i++){
            vector<int>v(i+1);
            for(int j=0;j<=i;j++){
                if(j==0 || j==i) v[j]=1;
                else v[j]=ans[i-1][j-1] +ans[i-1][j];
            }
            ans.push_back(v);
        }
        return ans;
    }
};